ocaml/xcp-rrdd/bin/rrdp-iostat/rrdp_iostat.ml

```diff
@@ -232,41 +232,51 @@ module Stat = struct
   type t = int64 list
   (* /sys/block/tdX/stats @ /sys/block/tdX/inflight @ [read bytes; write bytes] *)
 
+  let finally = Xapi_stdext_pervasives.Pervasiveext.finally
+
+  (** open [path] to let [f] scan the contents using scanf *)
+  let scanning path f =
+    let io = Scanf.Scanning.open_in path in
+    finally (fun () -> f io) (fun () -> Scanf.Scanning.close_in io)
+
+  let sector_size dev =
+    let path = String.concat "/" ["/sys/block"; dev; "queue/hw_sector_size"] in
+    try
+      scanning path @@ fun io ->
+      Scanf.bscanf io " %Ld" @@ fun d1 -> d1
+    with e ->
+      D.error "%s: %s" __FUNCTION__ (Printexc.to_string e) ;
+      0L
+
+  (** read first 11 values from device stat *)
+  let stat dev =
+    let path = String.concat "/" ["/sys/block"; dev; "stat"] in
+    try
+      scanning path @@ fun io ->
+      Scanf.bscanf io " %Ld %Ld %Ld %Ld %Ld %Ld %Ld %Ld %Ld %Ld %Ld"
+      @@ fun d1 d2 d3 d4 d5 d6 d7 d8 d9 d10 d11 ->
+      [d1; d2; d3; d4; d5; d6; d7; d8; d9; d10; d11]
+    with e ->
```

Comment:
when would this error occur, is there an example?

Reply:
The exception would occur if `path` can't be opened or if the the content does not match the expected format.